### PR TITLE
Upgrade to secure version of Django

### PR DESCRIPTION
**Pull request by marcpareoee**
- Disable Swagger since the API changed 
- Tweak migrations because they fail after upgrade because of a circular
  dependency (not clear why this worked in a previous version) -- will test upgrading the demo datastores to make sure the fix doesn't fail on an instance that has already applied some migrations
- Deleting Traces in tests caused an integrity error in tests. Wrap
  tests that do this in a transaction to prevent the tearDown from
failing from this.

```Traceback (most recent call last):
  File "/home/vagrant/venvs/datastore/local/lib/python2.7/site-packages/django/test/testcases.py", line 216, in __call__
    self._post_teardown()
  File "/home/vagrant/venvs/datastore/local/lib/python2.7/site-packages/django/test/testcases.py", line 908, in _post_teardown
    self._fixture_teardown()
  File "/home/vagrant/venvs/datastore/local/lib/python2.7/site-packages/django/test/testcases.py", line 1064, in _fixture_teardown
    connections[db_name].check_constraints()
  File "/home/vagrant/venvs/datastore/local/lib/python2.7/site-packages/django/db/backends/postgresql/base.py", line 224, in check_constraints
    self.cursor().execute('SET CONSTRAINTS ALL IMMEDIATE')
  File "/home/vagrant/venvs/datastore/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/vagrant/venvs/datastore/local/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/vagrant/venvs/datastore/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 62, in execute
    return self.cursor.execute(sql)
IntegrityError: insert or update on table "datastore_tracerecord" violates foreign key constraint "datastore_tracerecord_trace_id_2fec08cf_fk_datastore_trace_id"
DETAIL:  Key (trace_id)=(33) is not present in table "datastore_trace".```

@philngo any idea why there is a foreign key error from the DELETE? Would rather fix the underlying problem than apply the band-aid solution here.

### CODE REVIEW

**Comment by marcpareoee** [comment]
@philngo Mind looking at the changes closely? Perhaps there are better ways to fix the things I encountered.

I can probably fix Swagger, though considering just leaving it out altogether (is it used anywhere?)

I'm thinking of testing my rolling out on a clone and running meters, just to be sure that nothing surprising broke.

File: api_doc/urls.py [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+# from rest_framework_swagger.views import SwaggerApiView
 
 
 urlpatterns = [
-    url(r'^$', MySwaggerUIView.as_view(), name="django.swagger.base.view"),
-    url(r'^api-docs/$', MySwaggerResourcesView.as_view(),
-        name="django.swagger.resources.view"),
-    url(r'^api-docs/(?P<path>.*)/?$', SwaggerApiView.as_view(),
-        name='django.swagger.api.view'),
+    # url(r'^$', MySwaggerUIView.as_view(), name="django.swagger.base.view"),
```
Comment (philngo) [inline api_doc/urls.py]: Totally fine with ditching the whole api_doc app. I don't think anyone is using the auto-generated API docs. A recent DRF release has big enough updates on this that it's not worth trying to keep the existing method around.

File: auditing/tests/test_project_with_meter_result_should_be_in_projectsummarymart.py [hunk 2]
```diff
@@ -20,6 +23,7 @@ def test_task(self):
             trace=trace0,
             project=project0
         )
+
         with connection.cursor() as cursor:
             cursor.execute(
                 '''DELETE FROM datastore_trace WHERE trace_id = %s''',
```
Comment (philngo) [inline auditing/tests/test_project_with_meter_result_should_be_in_projectsummarymart.py]: I bet we can avoid the transactions if we also delete the orphaned trace records.

File: datastore/migrations/0038_auto_20161005_2208.py [hunk 3]
```diff
@@ -111,9 +110,9 @@ class Migration(migrations.Migration):
             model_name='project',
             name='active_project_result',
         ),
-        migrations.DeleteModel(
-            name='ConsumptionMetadata',
-        ),
+        # migrations.DeleteModel(
```
Comment (philngo) [inline datastore/migrations/0038_auto_20161005_2208.py]: What happened with these old migrations?

File: requirements.txt [hunk 4]
```diff
@@ -5,7 +5,7 @@ boto==2.46.1
 celery==3.1.25 # pyup: <4.0.0 # breaks tests
 kombu==3.0.37 # pyup: <4.0.0 # breaks tests
 dj-database-url==0.4.2
-Django==1.9.12 # pyup: <1.10.0 # breaks tests
+Django==1.10.7 # pyup: <1.11.0 # breaks tests
```
Comment (philngo) [inline requirements.txt]: Did upgrading to 1.11 look tricky?

File: auditing/tests/test_project_with_meter_result_should_be_in_projectsummarymart.py (hunk 2 above)
Comment (philngo) [inline auditing/tests/test_project_with_meter_result_should_be_in_projectsummarymart.py]: Even better, we can set up the `datastore_tracerecord` table with an `ON DELETE CASCADE`. I don't think there was any reason we were avoiding setting this explicitly.

File: requirements.txt (hunk 4 above)
Comment (philngo) [inline requirements.txt]: At least locally, it appears not to break anything.

**Comment by marcpareoee** [comment]
@philngo Turns out if you use ORM methods to delete instead of raw sql, Django takes care of the cascading delete. (Learned this after reading the docs more closely for on_delete behavior).

**Comment by philngo** [comment]
Good point! I think I tried using ORM deletes when I was writing the tests,
but then realized they were a bit more eager than I wanted in this
particular context, since those tests are specifically designed to mimic
what happens when traces are deleted in bulk using raw SQL and orphans are
left in the database. (The warehouse should ignore meter results for
deleted traces.) In a way, the ORM delete was too clean.

Now that I'm thinking about it, most of this will end up being moot pretty
soon since things will look a bit different with the meter manifests. Given
that, I'm ok with leaving the tests as is with the transactions and not
spending too many cycles on it. I will probably touch them again by the end
of the week anyway.

**Pull request merged**